app/code/Magento/MediaGalleryUi/Model/Directories/GetFolderTree.php

```diff
@@ -99,6 +114,29 @@ private function getDirectories(): array
         return $directories;
     }
 
+    /**
+     * Read only directories from file system
+     *
+     * @param string $pattern
+     * @param int $flags
+     */
+    private function recursiveRead(string $pattern, int $flags = 0): array
+    {
+        // phpcs:ignore Magento2.Functions.DiscouragedFunction
+        $directories = glob($pattern, $flags);
```

Comment:
Can you use \Magento\Framework\Filesystem\DriverInterface::search instead?

Reply:
@le0n4ik this would be great, but main idea is the flag to read directories only,  and search method have no ability to set this flag, and it will search only for one level.

Reply:
bu thanks, we van use Magento\Framework\Filesystem\Glob as well